stdlib/public/Concurrency/Task.swift

```diff
@@ -0,0 +1,273 @@
+////===----------------------------------------------------------------------===//
+////
+//// This source file is part of the Swift.org open source project
+////
+//// Copyright (c) 2020 Apple Inc. and the Swift project authors
+//// Licensed under Apache License v2.0 with Runtime Library Exception
+////
+//// See https://swift.org/LICENSE.txt for license information
+//// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+////
+////===----------------------------------------------------------------------===//
+
+import Swift
+@_implementationOnly import _SwiftConcurrencyShims
+
+// ==== Task -------------------------------------------------------------------
+
+/// An asynchronous task (just "Task" hereafter) is the analogue of a thread for
+/// asynchronous functions. All asynchronous functions run as part of some task.
+///
+/// A task's execution can be seen as a series of periods where the task was
+/// running. Each such period ends at a suspension point or -- finally -- the
+/// completion of the task.
+///
+/// These partial periods towards the task's completion are `PartialAsyncTask`.
+/// Partial tasks are generally not interacted with by end-users directly,
+/// unless implementing a scheduler.
+public struct Task {
```

Comment:
We should probably put an `OpaquePointer` or perhaps an `AnyObject` in here and make it frozen, because `Task` is a thin wrapper around a pointer to a task, which is a ref-counted entity.

Reply:
Ah true, will address in next PR 👍

Reply:
It's swift ref-counted, you should use `Builtin.NativeObject`.

Reply:
In the proposal this was just a namespace.  If we're going to make it a type that's a generic reference to a task, I think that's useful, but we need to think about how this interacts with `Handle`.  Maybe it's a class.

Reply:
The proposal indeed at first declared `enum Task` but also has a few `struct Task` but I guess those were always meant to be `extension Task` -- I guess it boils down to the primary question if one should be able to "touch" someone else's task ever at all.

We mentioned that if we allow `Task.current` one could potentially store or send it around; This interacts with things like task local storage and the deadline/timeout APIs...

Or should we stick to "task is a concept" and all interaction with it is via static functions? It is more correct I guess, i.e. we can then easily disallow any other task being able to set/get task local storage (if it was an API on the task object it could be abused).

So... back to namespace or there's reason to have it a class?